shared/util/use-focus-blur.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
+import {useNavigationEvents} from './navigation-hooks'
+import {NavigationEventCallback} from '@react-navigation/core'
+
+/**
+ * useFocusBlur sets up callbacks for when your component is focused or blurred.
+ * These happen when you are mounted/unmounted, and when navigation focus and
+ * blur events happen. onFocus and onBlur should be memoized, if not they may be
```

Comment:
> onFocus and onBlur should be memoized
Open to suggestions on better ways to handle this.

Reply:
I guess it's consistent with other hooks. WDYT of adding an eslint rule that notices if you pass unmemoized things into this, a la the one for React.useEffect?

Reply:
The alternate idea I can think of is passing in `deps` and calling `useCallback` on the first 2 args with the deps. But that seems a recipe for unneeded work so I think as is is fine, with the lint rule if we can swing it.

Reply:
I made it store the callbacks in refs and access those when focus/blur happens. it seems to work now with unmemoized closures.